oauth2_http/javatests/com/google/auth/oauth2/ComputeEngineCredentialsTest.java

```diff
@@ -250,6 +277,29 @@ public void getRequestMetadata_hasAccessToken() throws IOException {
     TestUtils.assertContainsBearerToken(metadata, accessToken);
   }
 
+  @Test
+  public void getRequestMetadata_shouldInvalidateAccessTokenWhenScoped_newAccessTokenFromRefresh()
+      throws IOException {
+    String accessToken = "1/MkSJoj1xsli0AccessToken_NKPY2";
+    String accessTokenWithScopes = "fake access token with scope";
+    MockMetadataServerTransportFactory transportFactory = new MockMetadataServerTransportFactory();
+    transportFactory.transport.setAccessToken(accessToken);
+    transportFactory.transport.setAccessToken("foo,bar", accessTokenWithScopes);
+    ComputeEngineCredentials credentials =
+        ComputeEngineCredentials.newBuilder().setHttpTransportFactory(transportFactory).build();
+    Map<String, List<String>> metadata = credentials.getRequestMetadata(CALL_URI);
+
+    TestUtils.assertContainsBearerToken(metadata, accessToken);
+
+    assertNotNull(credentials.getAccessToken());
+    ComputeEngineCredentials scopedCredentialCopy =
+        (ComputeEngineCredentials) credentials.createScoped(Arrays.asList("foo", "bar"));
+    assertNull(scopedCredentialCopy.getAccessToken());
+    Map<String, List<String>> metadataForCopiedCredentials =
+        scopedCredentialCopy.getRequestMetadata(CALL_URI);
+    TestUtils.assertContainsBearerToken(metadataForCopiedCredentials, accessTokenWithScopes);
+  }
```

Comment:
nit: I think one last assert which is assert that metadataForCopiedCredentials's access token does not equal metadata's accessToken.

It would be that `1/MkSJoj1xsli0AccessToken_NKPY2` != `fake access token with scope`